source/linux/Foundational_Components/U-Boot/UG-QSPI.rst

```diff
@@ -1,32 +1,232 @@
+
+   +------------+--------------+------------------------------------+
+   | SoC Family | Capability   | Driver                             |
+   +============+==============+====================================+
+   |            | 1x OSPI NOR  | :file:`drivers/spi/cadence_qspi.c` |
+   + J722S      +--------------+------------------------------------+
+   |            | 1x OSPI NAND | :file:`drivers/spi/cadence_qspi.c` |
+   +------------+--------------+------------------------------------+
+
+.. ifconfig:: CONFIG_part_variant in ('J721S2', 'J784S4','J742S2')
+
+   +---------------+--------------+------------------------------------+
+   | SoC Family    | Capability   | Driver                             |
+   +===============+==============+====================================+
+   |               | 1x QSPI NOR, | :file:`drivers/spi/cadence_qspi.c` |
+   | J721S2/J784S4 | 1x OSPI NOR  |                                    |
+   +               +--------------+------------------------------------+
+   |               | 1x OSPI NAND | :file:`drivers/spi/cadence_qspi.c` |
+   +---------------+--------------+------------------------------------+
 
 .. note::
 
-    The sf command is used to access SPI NOR flash, supporting read/write/erase and
-    a few other functions. For more information on sf command in U-boot please
-    refer to the u-boot documentation: `here <https://u-boot.readthedocs.io/en/latest/usage/cmd/sf.html>`__.
-    And for accessing SPI NAND flash, the mtd command is used, supporting
-    read/write/erase and bad block management.
+   Not all OSPI flashes can be supported. Users are recommended to check
+   whether or not the OSPI flash part chosen for custom board designs meets all
+   the criteria listed at https://e2e.ti.com/support/processors/f/791/t/946418
+
+.. rubric:: Driver Features
+
+OSPI controllers supports Double Data Rate (DDR) mode for OSPI/QSPI NOR flashes
+in Octal configuration wherein data can be read on both edges of the clock, and
+Single Data Rate (SDR) mode for OSPI/QSPI NAND flashes in Quad and Octal
+configuration.
+
+.. rubric:: Memory mapped read support
+   :name: u-boot-memory-mapped-read-support
+
+Once the controller is configured in memory map mode, the whole
+flash memory is available as a memory region at SoC specific address.
+This region can be accessed using normal memcpy() (or mem-to-mem dma
+copy). Controller hardware will internally communicate with
+SPI flash over SPI bus and get the requested data. This mode provides
+the best throughput and is the default mode in the SDK.
+
+.. rubric:: Supported SPI modes
+   :name: u-boot-supported-spi-modes
+
+:file:`cadence_qspi.c` driver supports standard SPI mode 0 only.
+
+.. rubric:: DMA support
+   :name: u-boot-dma-support
+
+Driver uses mem-to-mem DMA copy on top of OSPI/QSPI memory mapped port during
+read from flash for maximum throughput and reduced CPU load.
+
+OSPI does not support interfacing with non flash SPI slaves.
```

Comment:
The line says, OSPI in general doesnot support non flash SPI slaves ? Is it the Controller we donot support on our board? I see people drive displays with #[OSPI](https://community.st.com/t5/interface-and-connectivity-ics/stm32u5-ospi-gpdma-driving-an-amoled-display-gram-with-qspi/td-p/83517)